src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilter.java

```diff
@@ -60,6 +61,11 @@
      */
     private boolean optional;
 
+    /**
+     * Control if only consider added lines in file.
+     */
+    private boolean strategy;
```

Comment:
Why `boolean`? Please read the doc and check values this field is supposed to have

Reply:
done

Reply:
Please clarify where did you find `strategy = "all"` and `"added".equals(strategy)` strategies in the doc?

Reply:
>Based on the aforementioned cases the following strategies should be considered during implementation:
> line - consider only those violations which were occurred in new code lines only
>context - consider a wider context when new code introduces violations outside of new/changed lines, but let's >presume it is on the same branch in AST but on upper level.
>column - distinguish between existing and new violations on the changed code lines

Reply:
Please show me in this sentence from the doc where did you find `changed lines both belong to line strategy`:
> line - consider only those violations which were occurred in new code lines only

Reply:
> * context - consider a wider context when new code introduces violations outside of new/changed lines, 
according to this sentence, context is about violations outside of new/changed lines. So I think changed lines belongs to line strategy, and 

> * line - consider only those violations which were occurred in new code lines only

this line strategy definition is not proper.

I understand this example belongs to line strategy:

<img width="1049" alt="image" src="https://user-images.githubusercontent.com/50866227/87215248-50087200-c367-11ea-92f5-338b332152f1.png">

how about this one, I also think them belong to line strategy, because we violations are introduced inside of new/changed lines, not satisfy context's condition——`context - consider a wider context when new code introduces violations outside of new/changed lines`:

<img width="1285" alt="image" src="https://user-images.githubusercontent.com/50866227/87215254-60b8e800-c367-11ea-9446-664022c535b2.png">